OneCommunityGlobal/HGNRest
# Parthj unit tests and integration test for time zone api controller.js

# Description
This PR contains both unit tests and integration tests for the `timeZoneAPIController.js`. 

## Related PRS (if any):
- This pull request is not related to any frontend pull request.
- This PR consists of similar unit tests as [PR 1027](https://github.com/OneCommunityGlobal/HGNRest/pull/1027) but in addition this PR covers the scenario when the `.env file` consists of both `TIMEZONE_COMMON_KEY` and `TIMEZONE_PREMIUM_KEY` API keys.

## Main changes explained:
- This unit tests which were part of [PR 1027](https://github.com/OneCommunityGlobal/HGNRest/pull/1027), were not correct since they did not consider the scenario when the `.env file` already consisted of both `TIMEZONE_COMMON_KEY` and `TIMEZONE_PREMIUM_KEY` API keys. Hence the unit tests have been refined. 
- The `timeZoneAPIController.js` too is modified to read the API keys within the `timeZoneAPIController function`.

## How to test:
1. check into current branch i.e. `git checkout parthj_unit_tests_and_integration_test_for_timeZoneAPIController.js`
2. do `npm install`
3. Go to [OpenCage Website](https://opencagedata.com/) and either login or sign up.
4. Create a GeoCoding API key. You do not need to pay anything as we can perform testing using the free trail's API key.
5. Open `.env file`
6. Set `TIMEZONE_PREMIUM_KEY=some_random_string`
7. Set `TIMEZONE_COMMON_KEY=Generated_API_Key_Of_Step_4`
8. `Save .env`
9. run the test i.e. `npm run test timeZoneAPI`

## Screenshots or videos of changes:
\
`Final Result after successful execution of tests:`

<img width="921" alt="Screenshot 2024-08-02 at 2 59 13 PM" src="https://github.com/user-attachments/assets/0234aa36-91f0-4461-b524-9e8f60cb489c">

\
\
`Video Reference to generate API Key`

https://github.com/user-attachments/assets/e130939c-c81a-4c82-849c-9d709fbf5b3d
